modules/core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/DBCheck.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * (C) Copyright 2021 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+package org.nuxeo.ecm.core.storage;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.common.utils.TextTemplate;
+import org.nuxeo.launcher.commons.DatabaseDriverException;
+import org.nuxeo.launcher.config.ConfigurationConstants;
+import org.nuxeo.launcher.config.ConfigurationException;
+import org.nuxeo.launcher.config.ConfigurationHolder;
+import org.nuxeo.launcher.config.backingservices.BackingChecker;
+
+/**
+ * @since 9.2
+ * @apiNote reworked in 11.5
+ */
+public class DBCheck implements BackingChecker {
+
+    private static final Logger log = LogManager.getLogger(DBCheck.class);
+
+    protected static final List<String> DB_EXCLUDE_CHECK_LIST = Arrays.asList("default", "none");
+
+    protected static final String PARAM_DB_DRIVER = "nuxeo.db.driver";
+
+    protected static final String PARAM_DB_JDBC_URL = "nuxeo.db.jdbc.url";
+
+    protected static final String PARAM_DB_HOST = "nuxeo.db.host";
+
+    protected static final String PARAM_DB_PORT = "nuxeo.db.port";
+
+    protected static final String PARAM_DB_NAME = "nuxeo.db.name";
+
+    protected static final String PARAM_DB_USER = "nuxeo.db.user";
+
+    protected static final String PARAM_DB_PWD = "nuxeo.db.password";
+
+    @Override
+    public boolean accepts(ConfigurationHolder configHolder) {
+        return !DB_EXCLUDE_CHECK_LIST.contains(configHolder.getProperty(ConfigurationConstants.PARAM_TEMPLATE_DBTYPE));
+
+    }
+
+    @Override
+    public void check(ConfigurationHolder configHolder) throws ConfigurationException {
+        try {
+            checkDatabaseConnection(configHolder);
+        } catch (DatabaseDriverException e) {
+            log.debug(e, e);
+            log.error(e.getMessage());
+            throw new ConfigurationException("Could not find database driver: " + e.getMessage());
```

Comment:
why not showing the full stack trace on error, not only with debug logs? (same just below)

Reply:
I think this is because we want a nice logging in the console, as we call this code through nuxeoctl. We get the stack in debug mode when we pass the --debug option.